multidataset_plugin.cc

```diff
@@ -0,0 +1,544 @@
+#include "multidataset_plugin.h"
+
+
+static multidataset_array *multi_datasets;
+/*
+static hid_t *dataset_recycle;
+static hid_t *memspace_recycle;
+static hid_t *dataspace_recycle;
+*/
+static int dataset_size;
+static int dataset_size_limit;
+
+/*
+static int dataset_recycle_size;
+static int dataset_recycle_size_limit;
+static int dataspace_recycle_size;
+static int dataspace_recycle_size_limit;
+static int memspace_recycle_size;
+static int memspace_recycle_size_limit;
+*/
+hsize_t total_data_size;
```

Comment:
How about moving all these global variables to a struct returned from the `init_multidataset()` call?